src/app/action-item/action-item.component.spec.ts

```diff
@@ -11,50 +11,113 @@ import {VstsPullRequest} from '../../domain/action-item';
+      const pr: any = { repository: {name: 'test'} };
+      pr.labels = [{name: 'DO NOT MERGE', active: true}];
+
+      const vstsItem = new VstsPullRequest(pr);
+      expect(vstsItem.do_not_merge).toBeTruthy('There should be an indication of not merging');
+    });
 
-  it('vsts items with tag should indicate not to merge them', () => {
-    const pr: any = { repository: {name: 'test'} };
-    pr.labels = [{name: 'DO NOT MERGE', active: true}];
+    it('vsts items with tag and title should indicate not to merge them', () => {
+      const pr: any = { repository: {name: 'test'} };
+      pr.labels = [{name: 'DO NOT MERGE', active: true}];
+      pr.title = '[do not merge] happy cat code change';
 
-    const vstsItem = new VstsPullRequest(pr);
-    expect(vstsItem.do_not_merge).toBeTruthy('There should be an indication of not merging');
+      const vstsItem = new VstsPullRequest(pr);
+      expect(vstsItem.do_not_merge).toBeTruthy('There should be an indication of not merging');
+    });
   });
 
-  it('vsts items with tag and title should indicate not to merge them', () => {
-    const pr: any = { repository: {name: 'test'} };
-    pr.labels = [{name: 'DO NOT MERGE', active: true}];
-    pr.title = '[do not merge] happy cat code change';
+  describe('for VstsRelease items', () => {
+    describe('building', () => {
+      it('should show as needing action if the release failed', () => {
+        const releaseInfo: any = { releaseDefinition: {name: 'test release'},
+          _links: {web: {href: 'www.test.release'} },
+          environments: [{status: 'failed'}] };
+
+        const vstsRelease = new VstsRelease(releaseInfo);
+        expect(vstsRelease.building).toBe(false);
+      });
+
+      it('should show as building if the release is in progress', () => {
+        const releaseInfo: any = { releaseDefinition: {name: 'test release'},
+          _links: {web: {href: 'www.test.release'} },
+          environments: [{status: 'inProgress'}] };
+
+        const vstsRelease = new VstsRelease(releaseInfo);
+        expect(vstsRelease.building).toBe(true);
+      });
+
+    });
+
+    describe( 'preDeployApproval', () => {
+      it('should show as building if it is in progress and no approval exists', () => {
+        const releaseInfo: any = { releaseDefinition: {name: 'test release'},
+          _links: {web: {href: 'www.test.release'} },
+          environments: [{status: 'inProgress', preDeployApprovals: []}] };
+
+        const vstsRelease = new VstsRelease(releaseInfo);
+        expect(vstsRelease.building).toBe(true);
+      });
+
+      it('should show as needing action if it is failed and no approval exists', () => {
+        const releaseInfo: any = { releaseDefinition: {name: 'test release'},
+          _links: {web: {href: 'www.test.release'} },
+          environments: [{status: 'failed', preDeployApprovals: []}] };
+
+        const vstsRelease = new VstsRelease(releaseInfo);
+        expect(vstsRelease.building).toBe(false);
```

Comment:
the title of the test indicates this is testing _needing action_, but the assertion is asserting `#building`? is the verbiage simple inaccurate or the assertion? i'm guessing j/ the verbiage?

Reply:
yeah, so, originally, items were either building or broken (needing action)... the idea of `needing action` for a release is overloaded and indidcated by the fact that it is `not building` - maybe we have to rethink how we mark things on the board?